src/audio.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2020  Joshua Minter & Patryk Seregiet
+ *
+ * This file is part of ReplaySorcery.
+ *
+ * ReplaySorcery is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ReplaySorcery is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ReplaySorcery.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "audio.h"
+#include "util/log.h"
+#include "util/memory.h"
+#include "pthread.h"
+#define SAMPLES_PER_CALLBACK 4096 //has to be power of 2
+static pthread_spinlock_t sampleGetLock;
```

Comment:
Please no globals. You're already passing in an object via userdata just have the spinlock on that.

Reply:
ok

Reply:
done